상단 Nav 개발 (퍼블리싱)

akswnd98 opened the pull request:
> feat: 상단 nav prototype

- 아직 기능은 들어가지 않았습니다. (레이아웃만)
- 성인 모드 스위치가 nav에 구현 되었습니다.

akswnd98 committed:
feat: 상단 nav prototype

아직 기능은 들어가지 않았습니다. (레이아웃만)

akswnd98 committed:
feat: 성인 모드 스위치

성인 컨텐츠를 볼지 말지 여부를 결정할 수 있는 스위치가
상단 nav에 제공됩니다.

akswnd98 committed:
fix: nav 경로 변경

nav가 src/pages에서 src/components로 옮겨졌습니다.

Roharui commented on src/components/Nav/index.tsx:
```diff
@@ -0,0 +1,35 @@
+      <div css={mainStyle}>
+        <Logo />
+        <Search />
+        <Buttons />
```
프로필 아이콘 기능은 어디에 있나요?

akswnd98 replied:
까먹었습니다.
추가해서 다시 커밋 올리겠습니다.

dalcon10028 commented on src/components/Nav/Buttons/AdultSwitch/index.tsx:
```diff
@@ -0,0 +1,87 @@
+export default function AdultSwitch() {
+  const [active, setActive] = useState(false);
+
+  const rootStyle = css`
+    width: 46px;
+    height: 26px;
+    position: relative;
+    border-radius: 13px;
+    &:hover {
+      cursor: pointer;
+    }
+    border: none;
+    display: block;
+    background-color: white;
+  `;
```
function 내부는 렌더링마다 동작하는 걸로 알고 있어서 런타임에 변경될 일 없는 값들은 function 밖으로 빠지는게 맞는것 같습니다 위로 뺄지 아래로 뺄지는 의논해보아요

akswnd98 replied:
아 맞네요.
한 수 배워갑니다 ㅎㅎ.
성능 부분 고민해보겠습니다.

dalcon10028 commented on src/assets/theme/theme.ts:
```diff
@@ -11,12 +11,18 @@ declare module '@mui/material/styles' {
     cta?: PaletteColorOptions | undefined;
     main?: PaletteColorOptions | undefined;
     secondary?: PaletteColorOptions | undefined;
     hover?: PaletteColorOptions | undefined;
+    button?: PaletteColorOptions | undefined;
```
hover, button등등 meaning이 모호해서 지영님이랑 의논해봐야할 것 같아요 책갈피 용으로 남겨둘게요

dalcon10028 commented on src/pages/Home/index.tsx:
```diff
@@ -1,3 +1,9 @@
-  return <div>Home</div>;
+  return (
+    <div>
+      <Nav />
```
Nav는 모든 페이지에 보이니(다시 렌더링 되면 안되니) App.tsx에 넣어주세요

akswnd98 replied:
넵

dalcon10028 commented on src/components/Nav/Buttons/index.tsx:
```diff
@@ -0,0 +1,44 @@
+import theme from '../../../assets/theme/theme';
+import AdultSwitch from './AdultSwitch';
+
+const rootStyle = css`
+  font-weight: bold;
```
이건 취향인 것 같아서 의논 해보아요
https://emotion.sh/docs/styled#changing-based-on-props
css={} attribute 없앨 수 있어서

akswnd98 replied:
저도 익숙해서 css={}으로 해서 쓰다가
인터페이스 분리가 styled가 더 좋은거 같아서 (decorator패턴)
왠만하면 styled로 쓰는게 좋을거 같습니다.

dalcon10028 reviewed (changes requested):
고생하셨습니다 function 만 확인해주세요

dalcon10028 commented on src/components/Nav/Buttons/index.tsx:
```diff
@@ -0,0 +1,44 @@
+  margin-left: 16px;
+`;
+
+export default function Buttons() {
```
이름이 너무 범용적이에요

akswnd98 replied:
인정합니다.

dalcon10028 commented on src/components/Nav/index.tsx:
```diff
@@ -0,0 +1,35 @@
+  align-items: center;
+`;
+
+export default function Nav() {
```
일관되게 fullname을 쓰면 좋겠어요
Navigation or NavigationBar ..

dalcon10028 commented on public/svgs/hash.svg:
```diff
@@ -0,0 +1,5 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+<g id="24 / symbols / hash">
+<path id="icon" fill-rule="evenodd" clip-rule="evenodd" d="M11 2H9L8.4 8H4.2L4 10H8.2L7.8 14H3.2L3 16H7.6L7 22H9L9.6 16H13.6L13 22H15L15.6 16H19.8L20 14H15.8L16.2 10H20.8L21 8H16.4L17 2H15L14.4 8H10.4L11 2ZM10.2 10L9.8 14H13.8L14.2 10H10.2Z" fill="#606060"/>
+</g>
+</svg>
```
icon은 src내에서 사용해야하니 /src/assets 어딘가에 있었으면 좋겠어요

akswnd98 committed:
fix: issue 반영

- 프로필 아이콘 기능 추가
- 랜더링 최적화 반영
- Nav -> Navigation으로 이름 변경
- css in js: css -> styled로 대부분 변경 (transform의 경우에는 어쩔 수 없이 css를 써야하는 가?)
- icon 위치 변경 (src/assets/svg로)

dalcon10028 commented on src/components/Navigation/Buttons/Avatar/index.tsx:
```diff
@@ -0,0 +1,63 @@
+import styled from '@emotion/styled';
+import { Avatar as MuiAvatar } from '@mui/material';
+import { styled as muiStyled } from '@mui/system';
+import { useState } from 'react';
```
스타일 적용을 방식을 하나로 좋겠는데 (styled, muiStyled중에 하나만 사용) 
`import { styled } from '@mui/material/styles';` 이거 위주로 사용하는 것 같아서
이걸로 통일하시는건 어떠신가요?

https://mui.com/material-ui/react-slider/#inverted-track

dalcon10028 reviewed (approved):
고생하셨습니다
스타일은 function 위에 쓰는걸로 정하신것 같아서 다음 PR에 맞춰서 변경할게요~
스타일 방식 통일만 부탁드립니다
